src/platforms/macos/macosutils.mm

```diff
@@ -42,11 +44,35 @@
   Q_ASSERT(appId);
 
   NSString* loginItemAppId =
-      QString("%1.login-item").arg(QString::fromNSString(appId)).toNSString();
-  CFStringRef cfs = (__bridge CFStringRef)loginItemAppId;
-
-  Boolean ok = SMLoginItemSetEnabled(cfs, startAtBoot ? YES : NO);
-  logger.debug() << "Result: " << ok;
+    QString("%1.login-item").arg(QString::fromNSString(appId)).toNSString();
+  
+  // For macOS 13 and beyond, register() and unregister() methods
+  // are used for managing login items since SMLoginItemSetEnabled() is deprecated.
+  // For versions prior to macOS 13, SMLoginItemSetEnabled() is used.
+  #if defined(__MAC_13_0) && (__MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_13_0)
```

Comment:
The #ifdef doesn't seem to make sense here, because this occurs at build time. We need to dynamically determine the version of the macOS on the user's machine, which is performed using `if (@available(macOS 13, *)) `